test/models/architecture_test.rb

```diff
@@ -74,4 +74,45 @@ class ArchitectureTest < ActiveSupport::TestCase
     architecture = Architecture.new :name => "../../etc/shadow"
     assert_equal "etc-shadow", architecture.bootfilename_efi
   end
+
+  test "create with multi names" do
+    valid_name_list.each do |name|
+      architecture = FactoryBot.build(:architecture, :name => name)
+      assert architecture.valid?, "Validation failed for create with valid name: '#{name}' length: #{name.length})"
+      assert_equal architecture.name, name
+    end
+  end
+
+  test "should not create with invalid names" do
+    invalid_name_list.each do |name|
+      architecture = FactoryBot.build(:architecture, :name => name)
+      refute architecture.valid?, "Validation succeeded for create with invalid name: '#{name}' length: #{name.length})"
+      assert_includes architecture.errors.keys, :name
+    end
+  end
+
+  test "update with multi names" do
+    architecture = FactoryBot.create(:architecture)
+    valid_name_list.each do |new_name|
+      architecture.name =  new_name
+      assert architecture.valid?, "Validation failed for update with valid name: '#{new_name}' length: #{new_name.length})"
```

Comment:
`assert_valid` can be used here

Reply:
we are going to loose the failure message as assert_valid does not accept message

Reply:
`assert_valid` has a generic failure message that should be as good as the one provided imho.

Reply:
I think the point was we can specify custom message for assert_valid which I've seen used elsewhere, if I had to chose between the two, I'd probably vote for shoulda_matches because of efficiency